runtime/src/main/java/org/corfudb/util/FileWatcher.java

```diff
@@ -91,8 +98,15 @@ private void poll() {
                         kind == StandardWatchEventKinds.ENTRY_CREATE ||
                         kind == StandardWatchEventKinds.ENTRY_DELETE)
                         && filename.toString().equals(file.getName())) {
-                    log.info("FileWatcher: file {} changed. Invoking handler...", filename);
-                    onChange.run();
+
+                    String curFileHash = getFileHash();
+                    if (!Objects.equals(curFileHash, this.lastFileHash)) {
```

Comment:
Shouldn't this be Strings.equals() objects will only return true if the objects are the same, not if the contents of the hashes are equal right?

Reply:
The compared Strings are the hashcode of the files, and Objects.equals() internally calls [String.equals()](Objects.equals) which returns true if the sequence of chars are the same. Am I missing anything here?